### ISSUE TITLE
SuccessfulTasks() always the same as SubmittedTasks()

### ISSUE THREAD

**Issue by bojanth** [alitto/pond#84]
Hello,

I've noticed that `SuccessfulTasks()` call always returns the total number of submitted tasks.
I think this is happening because tasks after the timeout will be counted as successful.

Relevant code:
`t()` in https://github.com/alitto/pond/blob/main/task.go#L72 will not return an error, so err argument to `updateMetrics()` will always be nil https://github.com/alitto/pond/blob/main/pool.go#L276

To reproduce:
```go
package main_test

import (
	"context"
	"github.com/alitto/pond/v2"
	"github.com/stretchr/testify/require"
	"testing"
	"time"
)

func TestPondTasksCount(t *testing.T) {
	pool := pond.NewPool(1)

	timeout, cancel := context.WithTimeout(context.Background(), 8*time.Second)
	defer cancel()
	group := pool.NewGroupContext(timeout)

	task := func() {
		time.Sleep(time.Second * 5)
	}

	for i := 0; i < 10; i++ {
		group.Submit(func() {
			task()
		})
	}
	err := group.Wait()
	require.ErrorIs(t, err, context.DeadlineExceeded)

	// Since one task takes 5 seconds and the timeout is set to 8 seconds,
	// only the first and the second task should be successful.
	submitted := pool.SubmittedTasks()
	successful := pool.SuccessfulTasks()
	require.Greater(t, submitted, successful) // Error: "10" is not greater than "10"
}

```

**Comment by alitto** [comment]
Hey @bojanth!

Nice catch!, I just released a fix this in [`v2.1.3`](https://github.com/alitto/pond/releases/tag/v2.1.3). Failed tasks were also not counted when using `SubmitErr` and the passed task returned an explicit error.

Please try it out and let me know if this works as expected now.

Thanks!

**Comment by bojanth** [comment]
Hi @alitto!

Sorry for the late response.
I've just tested the latest version (`v2.1.5`) and it looks good! 🎉 

Thank you!
